File: 37.py
from datetime import date
zelda_dict = { #creamos un diccionario con los títulos y fechas de lanzamientos
    "tloz":date(1986,2,21),
    "zelda ii":date(1987,1,14),
    "a link to the past":date(1991,11,21),
    "ocarina of time":date(1998,11,21),
    "majora's mask":date(2000,4,27),
    "the wind waker":date(2002,12,13),
    "four swords adventures":date(2004,3,18),
    "twilight princess":date(2006,11,19),
    "skyward sword":date(2011,11,18),
    "breath of the wild":date(2017,3,3),
    "tears of the kingdom":date(2023,5,12),
    "link's awakening":date(1993,6,6),
    "oracle of seasons":date(2001,2,27),
    "oracle of ages":date(2001,2,27),
    "the minish cap":date(2004,11,4),
    "phantom hourglass":date(2007,6,23),
    "spirit tracks":date(2007,12,7),
    "a link between worlds":date(2013,11,22),
    "triforce heroes":date(2015,10,22),
    "four swords":date(2001,11,20)
}
def years_between_zelda_titles(zelda_title_one,zelda_title_two):
    zelda_title_two = zelda_title_two.lower()
    zelda_title_one = zelda_title_one.lower()
    titles = list(zelda_dict.keys())
    date_one, date_two = 0,0 #creamos variables para guardar fechas
    if zelda_title_one in titles:#si el primer titulo se encuentra entre los permitidos
        date_one = zelda_dict[zelda_title_one] #le asociamos la fecha deseada
    else:
        print(f"tu primer título no es válido, verifica que se encuentre entre los siguientes {titles}")#le mostramos los títulos válidos
        return None
    if zelda_title_two in titles: #hacemos lo mismo con los el segundo título
        date_two = zelda_dict[zelda_title_two]
    else:
        print(f"tu segundo título no es válido, verifica que se encuentre entre los siguientes {titles}")
        return None
    diff = abs(date_two-date_one)
    return f"la diferencia entre los títulos es {diff.days//365} años y {diff.days%365} días, osea {diff.days} días en total"
years_between_zelda_titles("fesif","zelda ii") #primer titulo erroneo
years_between_zelda_titles("tloz","wind waker") #segundo título errorneo
print(years_between_zelda_titles("tloz","ocarina of time"))

File: 37.js
zeldaDates={
    "Tloz":new Date(1986,1,21),
    "Zelda ii":new Date(1987,0,14),
    "A link to the past":new Date(1991,10,21),
    "Ocarina of time":new Date(1998,10,21),
    "Majora's mask":new Date(2000,3,27),
    "The wind waker":new Date(2002,11,13),
    "Four swords adventures":new Date(2004,2,18),
    "Twilight princess":new Date(2006,10,19),
    "Skyward sword":new Date(2011,10,18),
    "Breath of the wild":new Date(2017,2,3),
    "Tears of the kingdom":new Date(2023,4,12),
    "Link's awakening":new Date(1993,5,6),
    "Oracle of seasons":new Date(2001,1,27),
    "Oracle of ages":new Date(2001,1,27),
    "The minish cap":new Date(2004,10,4),
    "Phantom hourglass":new Date(2007,5,23),
    "Spirit tracks":new Date(2007,11,7),
    "A link between worlds":new Date(2013,10,22),
    "Triforce heroes":new Date(2015,9,22),
    "Four swords":new Date(2001,10,20)};
const timeBetweenZeldaTitles = (titleOne,titleTwo)=>{
    const zeldaDates={
        "Tloz":new Date(1986,1,21),
        "Zelda ii":new Date(1987,0,14),
        "A link to the past":new Date(1991,10,21),
        "Ocarina of time":new Date(1998,10,21),
        "Majora's mask":new Date(2000,3,27),
        "The wind waker":new Date(2002,11,13),
        "Four swords adventures":new Date(2004,2,18),
        "Twilight princess":new Date(2006,10,19),
        "Skyward sword":new Date(2011,10,18),
        "Breath of the wild":new Date(2017,2,3),
        "Tears of the kingdom":new Date(2023,4,12),
        "Link's awakening":new Date(1993,5,6),
        "Oracle of seasons":new Date(2001,1,27),
        "Oracle of ages":new Date(2001,1,27),
        "The minish cap":new Date(2004,10,4),
        "Phantom hourglass":new Date(2007,5,23),
        "Spirit tracks":new Date(2007,11,7),
        "A link between worlds":new Date(2013,10,22),
        "Triforce heroes":new Date(2015,9,22),
        "Four swords":new Date(2001,10,20)
    };
    if(zeldaDates[titleOne]!=undefined&&zeldaDates[titleTwo]!=undefined){
        let first = zeldaDates[titleOne];
        let second = zeldaDates[titleTwo];
        let years = 0;
        let days = 0;
        if(second-first>=0){
            let reference = new Date(second.getFullYear(),first.getMonth(),first.getDate());
            if(second-reference>=0){
                console.log("mayor");
                years = reference.getFullYear()-first.getFullYear();
                days = (second-reference)/(1000*60*60*24);
            } else{
                console.log("menor");
                reference = new Date(second.getFullYear()-1,first.getMonth(),first.getDate());
                years = reference.getFullYear()-first.getFullYear();
                days = (second-reference)/(1000*60*60*24);
            }
            console.log(reference,second,first);
            return `Hay ${years} años y ${days} días de diferencia entre los títulos`;
        }else{
            let reference = new Date(first.getFullYear,second.getMonth(),second.getDate());
            if(first-reference>=0){
                years = reference.getFullYear()-second.getFullYear();
                days = (first-reference)/(1000*60*60*24);
            } else{
                reference = new Date(first.getFullYear()-1,second.getMonth(),second.getDate());
                years = reference.getFullYear()-second.getFullYear();
                days = (first-reference)/(1000*60*60*24);
            }
            return `Hay ${years} años y ${days} días de diferencia entre los títulos`;
        }
    }else{
        return "alguno de los titulos que elegiste no es válido"
    }
}
zeldaDates["A link between worlds"]